### ISSUE TITLE
Unable to find address even if exact match

### ISSUE THREAD

**Issue by michaelkrog** [komoot/photon#221]
I have difficulties with some adresses.

**Example with a Danish address**
Hejreskovvej 10A 3490 Kvistgård

Openstreetmap has no problem finding it:
https://www.openstreetmap.org/search?query=Hejreskovvej%2010A%203490%20Kvistg%C3%A5rd

But photon cannot find it:
http://photon.komoot.de/api/?q=Hejreskovvej%2010A%203490%20Kvistg%C3%A5rd

**Comment by karussell** [comment]
Maybe the reason is that it is just an address without a 'house' OSM-way?

**Comment by karussell** [comment]
The character 'A' is the problemhttp://photon.komoot.de/api/?q=Hejreskovvej%2010%203490%20Kvistg%C3%A5rd 

The housenumber seems to be somehow incorrectly feeded as e.g. this works and correctly contains '11c': http://photon.komoot.de/api/?q=M%C3%BCnchen%20meindlstrasse%2011c

So there are two issues: a feeding problem and a missing fallback mechanism e.g. from 10A to 10

**Comment by jnns** [comment]
The problem also occurs with the following address: 

> Malplaquetstraße **5A**, 13347 Berlin, Germany 

```
   {
   	"features": [],
   	"type": "FeatureCollection"
   }
```

> Malplaquetstraße **5**, 13347 Berlin, Germany

```
{
	"features": [{
		"geometry": {
			"coordinates": [13.3617884, 52.5496013],
			"type": "Point"
		},
		"type": "Feature",
		"properties": {
			"osm_id": 2103565442,
			"osm_type": "N",
			"country": "Germany",
			"osm_key": "place",
			"housenumber": "5",
			"city": "Berlin",
			"street": "Malplaquetstraße",
			"osm_value": "house",
			"postcode": "13347",
			"state": "Berlin"
		}
	}],
	"type": "FeatureCollection"
}
```

**Comment by christophlingg** [comment]
I assume `5A` is split into two terms: `5` + `A`

**Comment by hbruch** [comment]
@karussell what feeding problem do you think of? http://photon.komoot.de/reverse?lat=55.99982&lon=12.50915 returns housenumber 10A, so it is fed into the index(?). And if 5A is split into 5 + A, why does it work for http://photon.komoot.de/api/?q=M%C3%BCnchen%20meindlstrasse%2011a ?

**Comment by karussell** [comment]
storing data in the index can be different to the reverted index

> And if 5A is split into 5 + A, why does it work for

Good question - maybe lower vs. upper case?

**Comment by hbruch** [comment]
I think you are right. The housenumber field is indexed not_analyzed, so a house number with uppercase letter is indexed unchanged. However, when querying, the standard analyzer is used, which performs a lowercase transformation, so no match is found. Field housenumber should at least be analyzed with a lowercase filter. I'll give it a try, building upon #254 ...

**Comment by hbruch** [comment]
Sorry, this issue was not closed by #275 as I changed the type of housenumber back to keyword. 

I‘m working on a fix, which still needs some testing. It will either be a minimal one just lowercasing the suffix or a broader one, which handles whitespaces in housenumbers as well.
